src/FixedDataTableCellGroup.js

```diff
@@ -65,74 +69,155 @@ class FixedDataTableCellGroupImpl extends React.Component {
 
   componentWillMount() {
     this._initialRender = true;
+    this._staticCellArray = [];
   }
 
   componentDidMount() {
     this._initialRender = false;
   }
 
   render() /*object*/ {
-    var props = this.props;
-    var columns = props.columns;
-    var cells = new Array(columns.length);
-    var contentWidth = sumPropWidths(columns);
+    const props = this.props;
+    const columns = props.columns;
+    const columnGroupWidth = sumPropWidths(columns);
 
-    var isColumnReordering = props.isColumnReordering && columns.reduce(function (acc, column) {
+    const isColumnReordering = props.isColumnReordering && columns.reduce(function (acc, column) {
       return acc || props.columnReorderingData.columnKey === column.props.columnKey;
     }, false);
 
-    var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
-      var columnProps = columns[i].props;
-      var cellTemplate = columns[i].template;
-      var recyclable = columnProps.allowCellsRecycling && !isColumnReordering;
-      if (!recyclable || (
-        currentPosition - props.left <= props.width &&
-        currentPosition - props.left + columnProps.width >= 0)) {
-        var key = columnProps.columnKey || 'cell_' + i;
-        cells[i] = this._renderCell(
-          props.rowIndex,
-          props.rowHeight,
-          columnProps,
-          cellTemplate,
-          currentPosition,
-          key,
-          contentWidth,
-          isColumnReordering
-        );
-      }
-      currentPosition += columnProps.width;
-    }
-    var style = {
+    // get list of cells
```

Comment:
refactor coming up

virtualized cells and non virtualized cells are calculated differently
* v-cells have their offsets and cell positions calculated through `columnOffsets` and `columnsToRender` props
* nv-cells need their offsets to be calculated, which can be done here, since it's just a linear incremental iteration (which is the existing implementation). Additionally, to not introduce any breaking change, `allowCellsRecycling` can be used here, which means that columns out side the viewport will can also be rendered, avoiding remounts at the cost of extra cell being rendered.